q2_diversity/_beta/_visualizer.py

```diff
@@ -159,15 +172,51 @@ def beta_group_significance(output_dir: str,
                                  urllib.parse.quote_plus(str(group_id))))
         fig.clear()
 
-    result = result.to_frame().to_html(classes="table table-striped "
-                                       "table-hover")
-    result = result.replace('border="1"', 'border="0"')
+    result_html = result.to_frame().to_html(classes=("table table-striped "
+                                                     "table-hover"))
+    result_html = result_html.replace('border="1"', 'border="0"')
+
+    if pairwise:
+        pairwise_results = []
+        for group1_id, group2_id in itertools.combinations(groupings, 2):
+            pairwise_result = \
+                _get_pairwise_group_significance_stats(
+                    distance_matrix=distance_matrix,
+                    group1_id=group1_id,
+                    group2_id=group2_id,
+                    groupings=groupings,
+                    metadata=metadata,
+                    beta_group_significance_fn=beta_group_significance_fn,
+                    permutations=permutations)
+            pairwise_results.append([group1_id,
+                                     group2_id,
+                                     pairwise_result['sample size'],
+                                     pairwise_result['test statistic'],
+                                     pairwise_result['p-value']])
+        columns = ['Group 1', 'Group 2', 'Sample size',
+                   result['test statistic name'], 'p-value']
+        pairwise_results = pd.DataFrame(pairwise_results, columns=columns)
+        pairwise_results.set_index(['Group 1', 'Group 2'], inplace=True)
+        pairwise_results['q-value'] = multipletests(
+            pairwise_results['p-value'], method='fdr_bh')[1]
+        pairwise_results.sort_index(inplace=True)
+        pairwise_path = os.path.join(output_dir, 'pairwise.csv')
```

Comment:
It'd also be useful to include a "number of permutations" column in the pairwise results so that the p-values can be interpreted/reported without needing the full visualization and its global test results.